File: insightedge-core/src/main/scala/com/gigaspaces/spark/python/PythonHelper.scala
package com.gigaspaces.spark.python

import com.gigaspaces.spark.context.GigaSpacesSparkContext
import com.gigaspaces.spark.java.JavaGigaSpacesSparkContext
import org.apache.spark.SparkContext
import org.apache.spark.api.java.JavaSparkContext

/**
  * A helper class to instantiate PythonGigaSpacesSparkContext from python using Py4J.
  *
  * @author Oleksiy_Dyagilev
  */
class PythonHelper {

  def createPythonGigaSpacesSparkContext(sc: JavaSparkContext): PythonGigaSpacesSparkContext = {
    val javaGsSparkContext = new JavaGigaSpacesSparkContext(sc)
    new PythonGigaSpacesSparkContext(javaGsSparkContext)
  }

}

File: insightedge-core/src/main/java/org/insightedge/spark/pyhton/PythonHelper.java
package org.insightedge.spark.pyhton;

import org.apache.spark.api.java.JavaSparkContext;
import org.insightedge.spark.context.InsightEdgeSparkContext;

public class PythonHelper {

    public PyhtonGigaspaceSparkContext createPythonGigaSpacesSparkContext(JavaSparkContext jsc) {
        return new PyhtonGigaspaceSparkContext(new InsightEdgeSparkContext(jsc.sc()));
    }

}
